readthedocs/config/config.py

```diff
@@ -1369,15 +1369,22 @@ def search(self):
         return Search(**self._config['search'])
 
 
-def load(path, env_config):
+def load(path, env_config, config_file=None):
     """
     Load a project configuration and the top-most build config for a given path.
 
     That is usually the root of the project, but will look deeper. According to
     the version of the configuration a build object would be load and validated,
     ``BuildConfigV1`` is the default.
     """
-    filename = find_one(path, CONFIG_FILENAME_REGEX)
+    if config_file is None or config_file == "":
+        filename = find_one(path, CONFIG_FILENAME_REGEX)
+    else:
+        filename = os.path.join(path, config_file)
+        if not os.path.exists(filename):
+            raise ConfigError(
+                f".readthedocs.yml not found at {config_file}", CONFIG_FILE_REQUIRED
+            )
 
```

Comment:
```suggestion
        filename = os.path.join(path, config_file)
        if not os.path.exists(filename):
            raise ConfigError(
                f".readthedocs.yml not found at {config_file}", CONFIG_FILE_REQUIRED
            )

```

Didn't properly dedent this,  but I think we don't want magic here if we can avoid it :)